test/components/CalendarMonth_spec.jsx

```diff
@@ -1,11 +1,36 @@
 import React from 'react';
 import { expect } from 'chai';
-import { shallow } from 'enzyme';
+import { shallow, mount } from 'enzyme';
 import sinon from 'sinon-sandbox';
 import moment from 'moment';
+import { describeIfWindow } from '../_helpers/describeIfWindow';
 
 import CalendarMonth from '../../src/components/CalendarMonth';
 
+const MAX_WAIT_TIME_IN_MS = 500;
+
+export default function awaitChange(fn) {
+  const startTimestamp = Date.now();
+
+  return new Promise((resolve, reject) => {
+    const checkAndWait = () => {
+      try {
+        fn();
+        resolve();
+      } catch (e) {
+        if (Date.now() - startTimestamp > MAX_WAIT_TIME_IN_MS) {
+          reject(e);
+          return;
+        }
+
+        setTimeout(checkAndWait, 10);
```

Comment:
Sorry if i wasn’t clear - any usage of setTimeout at all, or anything based on the concept of waiting for some amount of time, is inherently brittle and shouldn’t be used. We need access to the actual async behavior so we can explicitly await on it.

Reply:
I am not sure what you are proposing. Could you describe in more concrete terms how this change might look like?

Reply:
Since you want to wait until `wrapper.instance().setMonthTitleHeightTimeout` has ran, at the very least, you'd want to poll until that had become null. However, in this case, I think you could do something like:
```jsx
let timeout = wrapper.instance().setMonthTitleHeightTimeout;
const stub = sinon.stub(wrapper.instance(), 'setMonthTitleHeightTimeout');
stub.get(() => timeout);
const nulled = new Promise((resolve) => {
  stub.set((value) => {
    if (timeout !== null && value === null) resolve();
    timeout = value;
  });
});
return nulled.then(() => {
  expect(whatever);
});
```

Reply:
Thanks, I think I get what you want now. Your solution is already coupled to the actual implementation, so a much more straight forward solution would be to just stub out `setTimeout` for the this test and make it synchronous. The asynchronicity is irrelevant for this test. This would reduce coupling to between the implementation and the tests but may result in some weird behavior if some other piece of code relies on setTimeout be asynchronous. Your call 😸

Reply:
Coupling tests to the implementation is fine as long as changing the impl in a way that breaks it, breaks the tests.

That includes stubbing setTimeout, which could certainly work.